extensions/amp-minute-media-player/0.1/amp-minute-media-player.js

```diff
@@ -0,0 +1,480 @@
+    // Host that serves player configuration and content redirects
+    this.preconnect.url('https://www.oo-syringe.com', onLayout);
+  }
+
+  /** @override */
+  buildCallback() {
+    // Get attributes, assertions of values, assign instance variables.
+    // Build lightweight DOM and append to this.element.
+
+    const {element} = this;
+
+    this.contentType_ = userAssert(
+      element.getAttribute('data-content-type'),
+      'The data-content-type must be specified for <amp-minute-media-player> %s',
+      element
+    );
+
+    this.contentId_ =
+      this.contentType_ != 'semantic'
+        ? userAssert(
+            element.getAttribute('data-content-id'),
+            'The data-content-id must be specified for %s ' +
+              'data-content-type in <amp-minute-media-player> %s',
+            this.contentType_,
+            element
+          )
+        : '';
+
+    this.initSemanticFields_();
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+
+    // Warn if the player does not have video interface support
+    this.readyTimeout_ /*** NECESSARY?? ***/ /** @type {number} */ = Services.timerFor(
+      window
+    ).delay(function() {
+      user().warn(
+        TAG,
+        'Did not receive ready callback from player, ' +
+          'ensure it has the videojs-amp-support plugin.'
+      );
+    }, 3000);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    //Define which layouts our element support
+
+    //Size-defined layouts: fixed, fixed height, responsive and fill
+    return isLayoutSizeDefined(layout);
+    //return layout == Layout.RESPONSIVE;
+  }
+
+  /**
+   * Init Semantic params
+   * @private
+   */
+  initSemanticFields_() {
+    const {element} = this;
+    this.scannedElement_ = element.getAttribute('data-scanned-element') || '';
+    this.tags_ = element.getAttribute('data-tags') || '';
+    this.minimumDateFactor_ =
+      element.getAttribute('data-minimum-date-factor') || '';
+    //console.log(element.getAttribute('data-scanned-element-type'));
+    this.scannedElementType_ =
+      element.getAttribute('data-scanned-element-type') || '';
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handleMPlayerMessage_(event) {
+    if (
+      !originMatches(
+        event,
+        this.iframe_,
+        'https://syringe.s3-us-west-2.amazonaws.com'
+      )
+    ) {
+      return;
+    }
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+    const data = objOrParseJson(eventData);
+    //console.log(data); DONT GET HERE
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    const eventType = data['event'];
+    //console.log(eventType);
+    if (eventType == 'ready') {
+      /*** NECESSARY?? ***/
```

Comment:
What part is necessary? It seems like `readyTimeout_` is to fail in case the expected message is not received, and this cancels the timeout.